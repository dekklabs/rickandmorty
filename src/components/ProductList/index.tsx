import React, { useContext, useEffect, useState } from 'react'
import { Row } from "react-bootstrap";
import { DataContext } from '../../utils/DataContext';
import Loading from '../Loading';
import ProductItem from '../ProductItem';

const ProductList = ({setCart}:any) => {

    // const personajes = useContext(DataContext)
    const {personajes} = useContext(DataContext)

    return (
        <Row className="mt-4">
            {
                Object.entries(personajes).length !== 0
                    ?
                    personajes.map((character:any) => (
                        <ProductItem
                            key={character.id}
                            character={character}
                            setCart={setCart}
                        />
                    ))
                    :
                    <Loading />
            }
        </Row>
    )
}

export default ProductList

interface Producto {
    id: Number,
    name: String,
    price: Number
}