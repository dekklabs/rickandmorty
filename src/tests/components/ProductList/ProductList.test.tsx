import React from 'react';
import { shallow } from 'enzyme'
import ProductList from '../../../components/ProductList'

describe('Pruebas en ProductList component', () => {

    let wrapper = shallow(<ProductList />);

    test('Debe mostrar <ProductList />', () => {
        expect(wrapper).toMatchSnapshot();
    })
})