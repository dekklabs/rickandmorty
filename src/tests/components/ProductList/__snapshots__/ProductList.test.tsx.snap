// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Card component Debe de agregarse un producto 1`] = `ShallowWrapper {}`;

exports[`Pruebas en ProductList component Debe mostrar <ProductList /> 1`] = `ShallowWrapper {}`;
